### PR TITLE
[ESIMD] Fixed a crash lowering __esimd_unpack_mask

### PR DESCRIPTION
**Pull request by DenisBakhvalov**

### CODE REVIEW

File: llvm/test/SYCLLowerIR/esimd_lower_crash_zext.ll [hunk 1]
```diff
@@ -0,0 +1,12 @@
+; NOTE: Assertions have been autogenerated by utils/update_test_checks.py
+; RUN: opt -LowerESIMD -S < %s | FileCheck %s
+
```
Comment (kbobrovs) [inline llvm/test/SYCLLowerIR/esimd_lower_crash_zext.ll:3]: please add a description what this test checks

**Review by cmc-rep — DISMISSED** [review]

File: llvm/test/SYCLLowerIR/esimd_lower_crash_zext.ll (hunk 1 above)
Comment (DenisBakhvalov) [reply llvm/test/SYCLLowerIR/esimd_lower_crash_zext.ll:3]: Done.

File: llvm/test/SYCLLowerIR/esimd_lower_crash_zext.ll [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+; This test checks that there is no crash while lowering 
+; __esimd_unpack_mask.
```
Comment (kbobrovs) [inline llvm/test/SYCLLowerIR/esimd_lower_crash_zext.ll:5]:
```suggestion
; __esimd_unpack_mask when its argument is an integer constant.
```

File: llvm/test/SYCLLowerIR/esimd_lower_crash_zext.ll (hunk 1 above)
Comment (kbobrovs) [reply llvm/test/SYCLLowerIR/esimd_lower_crash_zext.ll:3]: I committed slight clarification to speed up things, hope you don't mind.

**Review by kbobrovs — APPROVED** [review]

**Pull request merged**